DeliveryCompany.AppForDrivers/PackageServices.cs

```diff
@@ -0,0 +1,166 @@
+            if (!waybill.Any(x => x.Id == deliveredPackageId))
+            {
+                Console.WriteLine("There is no such package.");
+                return;
+            }
+            if (waybill.Any(x => x.Id == deliveredPackageId && x.State == StateOfPackage.DeliveredManually))
+            {
+                Console.WriteLine("Package has already been delivered.");
+                return;
+            }
+            if (waybill.Any(x => x.Id == deliveredPackageId && x.ModeWaybill == ModeOfWaybill.Automatic))
+            {
+                CheckPackage(deliveredPackageId, waybill, waybillItems);
+                return;
+            }
+            else
+            {
+                UpdatePackageStatus(deliveredPackageId, waybill, waybillItems);
+                return;
+            }
+        }
+
+        private void CheckPackage(int deliveredPackageId, List<Package> waybill, List<WaybillItem> waybillItems)
+        {
+            var package = GetPackage(deliveredPackageId);
+            if (package == null)
+            {
+                Console.WriteLine("There is no such package.");
+                return;
+            }
+            if (package.State == StateOfPackage.Received)
+            {
+                Console.WriteLine("Package has already been delivered.");
+                return;
+            }
+            else
+            {
+                UpdatePackageStatus(deliveredPackageId, waybill, waybillItems);
+                return;
+            }
+        }
+
+        private Package GetPackage(int deliveredPackageId)
+        {
+            using (var httpClient = new HttpClient())
+            {
+                var response = httpClient.GetAsync($@"http://localhost:10500/api/packages/{deliveredPackageId}").Result;
+                var responseText = response.Content.ReadAsStringAsync().Result;
+
+                if (response.StatusCode == HttpStatusCode.OK)
+                {
+                    var responseObject = JsonConvert.DeserializeObject<Package>(responseText);
+                    return responseObject;
+                }
+                else
+                {
+                    Console.WriteLine("Something went wrong");
+                    return null;
+                }
+            }
+        }
+
+        private void UpdatePackageStatus(int deliveredPackageId, List<Package> _waybill, List<WaybillItem> waybillItems)
+        {
+            var package = _waybill.FirstOrDefault(x => x.Id == deliveredPackageId);
+
+            package.State = StateOfPackage.DeliveredManually;
+            package.DeliveryDate = _fastForwardTimeProvider.Now;
+            var estimatedDeliveryTime = waybillItems
+                .Where(x => x.Package.Id == deliveredPackageId && x.TypeOfPackage == TypeOfPackages.ToBeDelivered)
+                .Select(x => x.EstimatedDeliveryTime)
+                .FirstOrDefault();
+
+            package.CourierRating = CountCouriersRating(package, estimatedDeliveryTime);
```

Comment:
I would rather put rating of the delivery in the core app (the one with WebAPI and database access).